libuavcan/test/util/avl_tree.cpp

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright (C) 2019 Theodoros Ntakouris <zarkopafilis@gmail.com>
+ * */
+
+#include <gtest/gtest.h>
+#include <uavcan/util/avl_tree.hpp>
+#include <uavcan/dynamic_memory.hpp>
+#include <uavcan/driver/system_clock.hpp>
+
+using uavcan::AvlTree;
+
+struct Entry{
+    int key;
+    int payload;
+
+    bool operator<(const Entry & other) const
+    {
+        return this->key < other.key;
+    }
+
+    bool operator>(const Entry & other) const
+    {
+        return this->key > other.key;
+    }
+
+    bool operator==(const Entry & other) const
+    {
+        return this->key == other.key;
+    }
+};
+
+/* OOM-Unsafe */
+inline Entry *makeEntry(uavcan::PoolAllocator<64 * 8, 64> *allocator, int key, int payload){
+    void *praw = allocator->allocate(sizeof(Entry));
+
+    Entry *e = new (praw) Entry();
+    UAVCAN_ASSERT(e);
+
+    e->key = key;
+    e->payload = payload;
+    return e;
+}
+
+/* Basic sanity checks */
+TEST(AvlTree, Sanity){
```

Comment:
A bit of a nit but I'm going to be pedantic about this in the bluesky branch: prefer many small tests over fewer large ones. Each test case should test a single thing. This greatly simplifies bisection of regressions when adding/changing things later. It also prevents the anti-pattern of building up a lot of state during a unit test which taxes a human's ability to understand the test and how it failed.

Reply:
The sanity checks tests if basic insert/removal operations are completed successfully. 

There are 2 other tests I'm working on which test:
- If rotations are performed properly
- Multiple entries with same keys